### DISCUSSION TITLE
zustand persist in nextjs14 inital data is not taken from LocalStorage state

### DISCUSSION THREAD

**Discussion by Ryjekk** [pmndrs/zustand#2331] [Q&A]
I am using zustand to persist some data, but always on the refresh page data is blinking and is load from initial state instead of localStorege state, what can I do to fix it.

similar issue here -> https://stackoverflow.com/questions/77730931/zustand-persist-in-nextjs14

store.ts
```
const useStore = create<
  TransactionsSliceInterface
>()(
  persist(
    (...a) => ({
      ...createTransactionSlice(...a),
    }),
    {
      name: 'lin-storage',
      skipHydration: true,
      storage: createJSONStorage(() => localStorage),
    }
  )
);
```

createTransactionSlice
```
export const createTransactionSlice: StateCreator<TransactionResponse & TransactionActions> = (
  set
) => ({
  hasInitializedStore: false,
  transactionsTotalSum: {
    summarySum: 0,
  },
  transactionsSummary: [],
  transactions: [],
  totalElements: 0,
  totalPages: 0,
  setTransactions: (data: TransactionResponse) => set({ ...data }),
});

```

Hydration component called in layout
```
import { useEffect } from 'react';
import useStore from '@/app/_store/store';

export default function Hydration() {
  useEffect(() => {
    useStore.persist.rehydrate();
  }, []);

  return null;
}
```

how i retire data from store
```
  const { transactions, totalPages, setTransactions } =
    useStore((state) => state);
```

Hydration is performed correctly but i always get this blinking moment of getting data from initialised store, not localstorege
 
 ###Anyone can help?
 
 
> EDIT: 
 
 i am able to omit this issue by implementing store following way + disable Hydration ->> But i do believe this is not right way to fix it [because first fetch of data get 500 error]
```
let useStore;

if (typeof window !== 'undefined') {
  useStore = create<TransactionsSliceInterface>()(
    persist(
      (...a) => ({
        ...createTransactionSlice(...a),
      }),
      {
        name: 'lin-storage',
        // skipHydration: true,
        storage: createJSONStorage(() => localStorage),
      }
    )
  );
}

export { useStore };
```

**Comment by dai-shi** [comment]
AFAIU, blinking is unavoidable with SSR in general. Only the solution I know is to make a component client only (with useEffect) and show skeleton, but it's probably not what you want.

Hope @dbritto-dev can help.

**Reply by elijaholmos** [reply]
I'm running into a similar issue with my Nextjs 14 app - reloading the page causes the store to populate with the default values for a second, then it retrieves & applies whatever vals were set in localStorage.

`'use client'` is present at the top of the component where this store is used. Ideally I would like to avoid skeletons

**Comment by dbritto-dev** [comment]
@Ryjekk btw, did you read our new guides, ssr and hydration + nextjs?
